### PR TITLE
Add GitHub App data to configuration

### PR DESCRIPTION
**Pull request by eleftherias**
The introduces a `provider` section in the configuration that allows configuring provider settings.

It also allows for optional fields in the config, by adding pointers to the allowed types.

### CODE REVIEW

File: internal/providers/providers.go [hunk 1]
```diff
@@ -88,14 +84,10 @@ func WithRestClientCache(cache ratecache.RestClientCache) ProviderBuilderOption
...
-func NewProviderBuilder(
-	p *db.Provider,
-	ownerFilter sql.NullString,
-	credential provinfv1.Credential,
-	opts ...ProviderBuilderOption,
-) *ProviderBuilder {
+func NewProviderBuilder(p *db.Provider, ownerFilter sql.NullString, credential provinfv1.Credential, cfg *serverconfig.Config, opts ...ProviderBuilderOption) *ProviderBuilder {
 	pb := &ProviderBuilder{
 		p:           p,
+		cfg:         cfg,
```
Comment (eleftherias) [inline internal/providers/providers.go:90]: We'd have to pass the config everywhere when we create a new `ProviderBuilder`.

File: internal/providers/providers.go [hunk 2]
```diff
@@ -210,7 +202,7 @@ func (pb *ProviderBuilder) GetGitHub() (provinfv1.GitHub, error) {
 		return nil, fmt.Errorf("error parsing github app config: %w", err)
 	}
 
-	cli, err := githubapp.NewGitHubAppProvider(cfg, pb.metrics, pb.restClientCache, gitHubCredential)
+	cli, err := githubapp.NewGitHubAppProvider(cfg, pb.cfg.GitHubApp, pb.metrics, pb.restClientCache, gitHubCredential)
```
Comment (eleftherias) [inline internal/providers/providers.go:205]: It seems weird to pass both the config from the definition and from the `config.yaml`. If we go down this route, should we also move the endpoint to the `config.yaml` and remove it from the provider definition?

File: internal/providers/github/app/app.go [hunk 3]
```diff
@@ -84,20 +85,26 @@ func NewGitHubAppProvider(
...
 		if err != nil {
 			return nil, err
 		}
 		ghClient.BaseURL = parsedURL
 	}
 
+	appName := appConfig.AppName
+	userId := appConfig.UserID
+
+	// Alternatively user viper to get the values directly
```
Comment (eleftherias) [inline internal/providers/github/app/app.go:99]: We also have the option of getting the values directly from viper but that makes testing harder.

**Review by eleftherias — COMMENTED** [review]
@JAORMX I added some comments about what this approach would look like.

File: internal/providers/providers.go (hunk 1 above)
Comment (JAORMX) [reply internal/providers/providers.go:90]: Yeah, I don't think there's away around that.... unless we end up creating a `ProviderBuilderFactory` that has a set of persistent values like metrics and configuration. (screams in Java)

File: internal/providers/providers.go (hunk 2 above)
Comment (JAORMX) [reply internal/providers/providers.go:205]: I'm not sure about the endpoint, could we have a minder instance that has tenants that use a private installation and other that point to the public github endpoint?

Comment (eleftherias) [reply internal/providers/providers.go:205]: I think that's a valid scenario. I also think we don't need to support it at this moment.

File: internal/config/server/config.go [hunk 4]
```diff
@@ -43,6 +43,7 @@ type Config struct {
 	WebhookConfig WebhookConfig         `mapstructure:"webhook-config"`
 	Events        EventConfig           `mapstructure:"events"`
 	Authz         AuthzConfig           `mapstructure:"authz"`
+	GitHubApp     GitHubAppConfig       `mapstructure:"github-app"`
```
[the remark this answers is not in the document]
Comment (eleftherias) [reply internal/config/server/config.go:46]: I like it

**Review by JAORMX — APPROVED** [review]

**Review by JAORMX — APPROVED** [review]

**Pull request merged**